rudder-web/src/main/scala/com/normation/rudder/web/components/RuleCompliance.scala

```diff
@@ -115,137 +113,177 @@ class RuleCompliance (
    * For each table : the subtable is contained in td : details
    * when + is clicked: it gets the content of td details then process it
    * as a datatable
+   *
+   * The table are empty when the page is displayed, then there is a JS call
+   * to refresh() that fill them.
    */
   def showCompliance : NodeSeq = {
+    <div id="directiveComplianceSection" class="unfoldedSection" onclick="$('#directiveCompliance').toggle(400); $('#directiveComplianceSection').toggleClass('foldedSection');$('#directiveComplianceSection').toggleClass('unfoldedSection');">
+      <div class="section-title">Compliance by Directive</div>
+    </div>
+    <div id="directiveCompliance">
+      <table id="reportsGrid" cellspacing="0">  </table>
+    </div>
+    <div id="nodeComplianceSection" class="unfoldedSection" onclick="$('#nodeCompliance').toggle(400); $('#nodeComplianceSection').toggleClass('foldedSection');$('#nodeComplianceSection').toggleClass('unfoldedSection');">
+      <div class="section-title">Compliance by Node</div>
+    </div>
+    <div id="nodeCompliance">
+       <table id="nodeReportsGrid" cellspacing="0">  </table>
+    </div>
+    <div id="recentChangesSection" class="unfoldedSection" onclick="$('#recentChanges').toggle(400); $('#recentChangesSection').toggleClass('foldedSection');$('#recentChangesSection').toggleClass('unfoldedSection');">
+      <div class="section-title">Recent changes</div>
+    </div>
+    <div id="recentChanges">
+      {SHtml.button("Refresh", () => refresh())}
+      <div id="changesChart">  </div>
+      <hr class="spacer" />
+      <table id="changesGrid" cellspacing="0">  </table>
+    </div> ++
+    Script(After(0,JsRaw(s"""
+      function refresh() {${refresh().toJsCmd}};
```

Comment:
So we discussed with françois, we keep the behavior like before, refresh the 3 parts then the refresh call is ok :)